crates/ruff_linter/src/rules/flake8_use_pathlib/rules/replaceable_by_pathlib.rs

```diff
@@ -86,7 +86,39 @@ pub(crate) fn replaceable_by_pathlib(checker: &mut Checker, expr: &Expr) {
                 // PTH205
                 ["os", "path", "getctime"] => Some(OsPathGetctime.into()),
                 // PTH123
-                ["" | "builtin", "open"] => Some(BuiltinOpen.into()),
+                ["" | "builtin", "open"] => {
+                    // `closefd` and `openener` are not supported by pathlib, so check if they are
+                    // are set to non-default values.
+                    // https://github.com/astral-sh/ruff/issues/7620
+                    // Signature as of Python 3.11 (https://docs.python.org/3/library/functions.html#open):
+                    // ```
+                    //      0     1         2             3              4            5
+                    // open(file, mode='r', buffering=-1, encoding=None, errors=None, newline=None,
+                    //      6             7
+                    //      closefd=True, opener=None)
+                    //              ^^^^         ^^^^
+                    // ```
+                    if call
+                        .arguments
+                        .find_argument("closefd", 6)
+                        .is_some_and(|expr| {
+                            !expr.as_constant_expr().is_some_and(|constant| {
+                                matches!(constant.value, Constant::Bool(true))
+                            })
+                        })
```

Comment:
Do we have a `.is_default_value("closefd", 6, Constant::Bool(true))` helper? If no, should i create one?

Reply:
I think what you have here is fine personally.